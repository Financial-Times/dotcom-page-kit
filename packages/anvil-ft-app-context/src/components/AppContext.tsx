import React from 'react'
import { TAppContext } from '../types'
import { APP_CONTEXT_ELEMENT_ID } from '../constants'

export interface TAppContextProps {
  data: TAppContext
}

export function AppContext({ data }: TAppContextProps) {
  return (
    <script
      type="application/json"
      id={APP_CONTEXT_ELEMENT_ID}
      dangerouslySetInnerHTML={{ __html: JSON.stringify(data) }}
    />
  )
}
