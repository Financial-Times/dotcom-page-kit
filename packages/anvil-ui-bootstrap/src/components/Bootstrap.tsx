import React from 'react'
import { formatConfigJSON, getBootstrapJS } from '../server'

type TBootstrapProps = {
  coreScripts?: string[]
  enhancedScripts?: string[]
}

function Bootstrap({ coreScripts, enhancedScripts }: TBootstrapProps) {
  return (
    <React.Fragment>
      <script
        type="application/json"
        id="anvil-bootstrap-config"
        dangerouslySetInnerHTML={{ __html: formatConfigJSON(coreScripts, enhancedScripts) }}
      />
      <script dangerouslySetInnerHTML={{ __html: getBootstrapJS() }} />
    </React.Fragment>
  )
}

Bootstrap.defaultProps = {
  coreScripts: [],
  enhancedScripts: []
}

export { Bootstrap, TBootstrapProps }
