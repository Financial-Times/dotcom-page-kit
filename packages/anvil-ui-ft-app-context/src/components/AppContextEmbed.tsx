import React from 'react'
import { TAppContext } from '../types'
import { APP_CONTEXT_ELEMENT_ID } from '../../constants'

export interface TAppContextProps {
  context: TAppContext
}

export function AppContextEmbed({ context }: TAppContextProps) {
  return (
    <script
      type="application/json"
      id={APP_CONTEXT_ELEMENT_ID}
      dangerouslySetInnerHTML={{ __html: JSON.stringify(context) }}
    />
  )
}

// test
