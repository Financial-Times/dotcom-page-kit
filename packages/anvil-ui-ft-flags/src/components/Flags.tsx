import React from 'react'
import { formatFlagsJSON } from '../server'
import { TFlagsData } from '../types'

type TFlagsProps = {
  flags?: TFlagsData
}

function Flags({ flags }: TFlagsProps) {
  return (
    <script
      type="application/json"
      id="anvil-flags-data"
      dangerouslySetInnerHTML={{ __html: formatFlagsJSON(flags) }}
    />
  )
}

Flags.defaultProps = {
  flags: {}
}

export { Flags, TFlagsProps }
