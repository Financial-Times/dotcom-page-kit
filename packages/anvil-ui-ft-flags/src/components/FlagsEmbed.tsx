import React from 'react'
import { formatFlagsJSON } from '../server'
import { TFlagsData } from '../types'
import { SCRIPT_ELEMENT_ID } from '../constants'

type TFlagsEmbedProps = {
  flags?: TFlagsData
}

function FlagsEmbed({ flags }: TFlagsEmbedProps) {
  return (
    <script
      type="application/json"
      id={SCRIPT_ELEMENT_ID}
      dangerouslySetInnerHTML={{ __html: formatFlagsJSON(flags) }}
    />
  )
}

FlagsEmbed.defaultProps = {
  flags: {}
}

export { FlagsEmbed, TFlagsEmbedProps }
