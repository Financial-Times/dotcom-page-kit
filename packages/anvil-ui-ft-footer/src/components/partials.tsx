import React from 'react'

const Link: Function = ({ item }) => {
  const disableTracking = { 'data-o-tracking-do-not-track': item.disableTracking ? 'true' : null }
  return (
    <a
      className="o-footer__matrix-link"
      href={`${item.url}`}
      data-trackable={`${item.label}`}
      {...disableTracking}>
      {item.label}
    </a>
  )
}

const SectionLinks = ({ submenu, index }) => {
  return (
    <div className="o-footer__matrix-content" id={`o-footer-section-${index}`}>
      {submenu.map((submenuItems, columnIndex) => (
        <div className="o-footer__matrix-column" key={`footer-section-${index}--column-${columnIndex}`}>
          {submenuItems.map((item, itemIndex) => {
            return <Link item={item} index={itemIndex} key={`footer-section-${index}--link-${itemIndex}`} />
          })}
        </div>
      ))}
    </div>
  )
}

const SectionTitle = ({ label, submenu, index }) => {
  // On smaller viewports, submenus which span two columns are collapsed behind toggles
  const ariaControls = { 'aria-controls': submenu.length === 2 ? `o-footer-section-${index}` : null }
  return (
    <div className="o-footer__matrix-title" {...ariaControls}>
      {label}
    </div>
  )
}

const MenuSection: Function = ({ label, submenu, index }) => {
  return (
    <div className={`o-footer__matrix-group o-footer__matrix-group--${submenu.length}`}>
      <SectionTitle label={label} submenu={submenu} index={index} />
      <SectionLinks submenu={submenu} index={index} />
    </div>
  )
}

const FooterRow = (props) => (
  <div className="o-footer__row">
    <nav className="o-footer__matrix" role="navigation" aria-label="Useful links">
      {props.children}
    </nav>
  </div>
)

const FooterContents = ({ footerData }) => {
  return footerData.map((item, index) => (
    <MenuSection
      label={item.label}
      submenu={item.submenu.items}
      index={index}
      key={`footer-section-${index}`}
    />
  ))
}

const MoreFromFt = () => (
  <div className="o-footer__external-link o-footer__matrix-title">
    <a
      className="o-footer__more-from-ft o-footer__matrix-title"
      href="http://ft.com/more-from-ft-group"
      data-trackable="more-from-ft">
      More from the FT Group
    </a>
  </div>
)

const CopyrightNotice: Function = ({ withoutMarketsData }) => {
  const marketsData = withoutMarketsData ? null : 'Markets data delayed by at least 15 minutes.'
  return (
    <div className="o-footer__copyright" role="contentinfo">
      <small>
        {marketsData} © The Financial Times LTD.
        <abbr title="Financial Times" aria-label="F T">
          FT
        </abbr>{' '}
        and ‘Financial Times’ are trademarks of The Financial Times Ltd.
        <br />
        The Financial Times and its journalism are subject to a self-regulation regime under the{' '}
        <a href="http://aboutus.ft.com/en-gb/ft-editorial-code/" aria-label="F T Editorial Code of Practice">
          FT Editorial Code of Practice
        </a>
        .
      </small>
    </div>
  )
}

const LegalLink: Function = ({ item, index }) => {
  const disableTracking = { 'data-o-tracking-do-not-track': item.disableTracking ? 'true' : null }
  return (
    <li key={`legalFooter-link-${index}`}>
      <a href={`${item.url}`} data-trackable={`${item.label}`} {...disableTracking}>
        {item.label}
      </a>
    </li>
  )
}

const CompressedLegal = ({ footerData }) => {
  const LegalAndPrivacy = footerData.filter((item) => {
    return item.label === 'Legal & Privacy'
  })
  return LegalAndPrivacy.map((legal, sectionIndex) => (
    <div key={`legalFooter-${sectionIndex}`}>
      {legal.submenu && legal.submenu.items
        ? legal.submenu.items.map((items, submenuIndex) => (
            <ul className="o-footer__legal-links" key={`legalFooter-submenu-${submenuIndex}`}>
              {items.map((item, submenuIndex) => {
                return <LegalLink item={item} key={`legalFooter-link-${submenuIndex}`} />
              })}
            </ul>
          ))
        : null}
    </div>
  ))
}

const NikkeiBrandLogo = () => (
  <div className="o-footer__brand">
    <div className="o-footer__container">
      <div className="o-footer__brand-logo" />
    </div>
  </div>
)

export {
  SectionLinks,
  SectionTitle,
  MenuSection,
  FooterRow,
  FooterContents,
  MoreFromFt,
  CompressedLegal,
  CopyrightNotice,
  NikkeiBrandLogo
}
