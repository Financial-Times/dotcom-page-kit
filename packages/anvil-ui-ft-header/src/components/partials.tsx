import React from 'react'

const HeaderHtml = (props) => (
  <header
    id="site-navigation"
    className="o-header"
    data-o-component="o-header"
    data-o-header--no-js={true}
    tabIndex={-1}>
    {props.children}
  </header>
)

const Drawer = () => (
  <a
    href="#o-header-drawer"
    className="o-header__top-link o-header__top-link--menu"
    aria-controls="o-header-drawer"
    title="Open drawer menu"
    aria-label="Open drawer menu"
    data-trackable="drawer-toggle">
    <span className="o-header__top-link-label">Menu</span>
  </a>
)

const Search = () => (
  <a
    href="#o-header-search-primary"
    className="o-header__top-link o-header__top-link--search"
    aria-controls="o-header-search-primary"
    title="Search"
    aria-label="Search"
    data-trackable="search-toggle">
    <span className="o-header__top-link-label">Search</span>
  </a>
)

const FooterTopWrapper = (props) => (
  <div className="o-header__row o-header__top" data-trackable="header-top">
    <div className="o-header__container">
      <div className="o-header__top-wrapper">{props.children}</div>
    </div>
  </div>
)

const TopColumnLeft = () => {
  return (
    <div className="o-header__top-column o-header__top-column--left">
      <Drawer />
      <Search />
    </div>
  )
}

const TopColumnCenter = () => (
  <div className="o-header__top-column o-header__top-column--center">
    <div className="o-header__top-logo">
      <span className="o-header__visually-hidden">Financial Times</span>
    </div>
  </div>
)

const MyFt = () => (
  <a
    className="o-header__top-link o-header__top-link--myft"
    href="/myft"
    data-trackable="my-ft"
    data-tour-stage="myFt"
    aria-label="My F T">
    <span className="o-header__visually-hidden">myFT</span>
  </a>
)

const TopColumnRight = () => {
  return (
    <div className="o-header__top-column o-header__top-column--right">
      <span className="o-header__top-link--myft__container">
        <MyFt />
      </span>
    </div>
  )
}

const NavListLeft = ({ navbarUk }) => {
  return (
    <ul className="o-header__nav-list o-header__nav-list--left" data-trackable="primary-nav">
      {navbarUk.map((navItem, index) => {
        const ariaLabel = navItem.selected ? { 'aria-label': 'Current page', 'aria-current': true } : null
        return (
          <li className="o-header__nav-item">
            <a
              className="o-header__nav-link o-header__nav-link--primary"
              href={navItem.href}
              id={`o-header-link-${index}`}
              {...ariaLabel}
              data-trackable={navItem.label}>
              {navItem.label}
            </a>
          </li>
        )
      })}
    </ul>
  )
}

const NavListRight = ({ navbarOption }) => {
  return (
    <ul className="o-header__nav-list o-header__nav-list--right" data-trackable="user-nav">
      {navbarOption.map((navItem) => {
        return (
          <li className="o-header__nav-item">
            <a className="o-header__nav-link" href={navItem.href} data-trackable={navItem.label}>
              {navItem.label}
            </a>
          </li>
        )
      })}
    </ul>
  )
}

const NavHtml = (props) => {
  return (
    <nav
      id="o-header-nav-desktop"
      className="o-header__row o-header__nav o-header__nav--desktop"
      role="navigation"
      aria-label="Primary navigation"
      data-trackable="header-nav:desktop">
      <div className="o-header__container">{props.children}</div>
    </nav>
  )
}

export {
  HeaderHtml,
  FooterTopWrapper,
  TopColumnLeft,
  TopColumnCenter,
  TopColumnRight,
  NavListLeft,
  NavListRight,
  NavHtml
}
