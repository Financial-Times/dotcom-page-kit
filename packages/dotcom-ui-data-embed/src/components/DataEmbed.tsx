import React from 'react'

export function DataEmbed({ id, data }: { id: string; data: object }) {
  return <script type="application/json" id={id} dangerouslySetInnerHTML={{ __html: JSON.stringify(data) }} />
}
