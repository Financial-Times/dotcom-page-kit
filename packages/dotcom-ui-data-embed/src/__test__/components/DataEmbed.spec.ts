import renderer from 'react-test-renderer'
import { DataEmbed as subject } from '../../components/DataEmbed'

const data = {
  id: 'data-embed',
  data: {
    foo: 1,
    bar: true,
    baz: 'qux'
  }
}

describe('dotcom-ui-data-embed/src/components/dataEmbed', () => {
  it('renders a script element containing data', () => {
    const tree = renderer.create(subject(data))
    expect(tree).toMatchSnapshot()
  })
})
