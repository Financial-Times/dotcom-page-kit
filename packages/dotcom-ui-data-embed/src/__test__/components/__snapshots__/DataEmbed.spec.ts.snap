// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dotcom-ui-data-embed/src/components/dataEmbed renders a script element containing data 1`] = `
<script
  dangerouslySetInnerHTML={
    {
      "__html": "{\"foo\":1,\"bar\":true,\"baz\":\"qux\"}",
    }
  }
  id="data-embed"
  type="application/json"
/>
`;
